thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -363,6 +369,46 @@ export default Ember.Controller.extend({
     };
   },
 
+  /**
+   * Fetches the current session state from the backend and issues a notification if it has been updated
+   *
+   * @private
+   */
+  _checkSession() {
+    const { sessionId, sessionUpdatedTime } = this.getProperties('sessionId', 'sessionUpdatedTime');
+
+    if (!sessionId) { return; }
+
+    fetch(`/session/${sessionId}`)
```

Comment:
can we try keeping all fetching logic out of the controllers? (Thank you for your effort in abstracting them to services btw!)